src/unity/python/turicreate/test/test_one_shot_object_detector.py

```diff
@@ -0,0 +1,290 @@
+
+        ann = []
+        for j in range(rs.randint(max_num_boxes_per_image)):
+            left, right = np.sort(rs.randint(0, img_shape[1], size=2))
+            top, bottom = np.sort(rs.randint(0, img_shape[0], size=2))
+
+            x = (left + right) / 2
+            y = (top + bottom) / 2
+
+            width = max(right - left, 1)
+            height = max(bottom - top, 1)
+
+            label = {
+                'coordinates': {
+                    'x': x,
+                    'y': y,
+                    'width': width,
+                    'height': height,
+                },
+                'label': classes[rs.randint(len(classes))],
+                'type': 'rectangle',
+            }
+            ann.append(label)
+        anns.append(ann)
+
+    starter_images = []
+    starter_target = []
+    for i in range(num_starter_images):
+        img_shape = tuple(rs.randint(100, 1000, size=2)) + (3,)
+        img = rs.randint(255, size=img_shape)
+        pil_img = _PIL_Image.fromarray(img, mode='RGB')
+        image_format = FORMATS[rs.randint(len(FORMATS))]
+        starter_images.append(from_pil_image(pil_img, image_format=image_format))
+        starter_target.append(i % len(_CLASSES))
+
+    train = tc.SFrame({
+        feature: tc.SArray(starter_images),
+        target: tc.SArray(starter_target),
+    })
+    test = tc.SFrame({
+        feature: tc.SArray(images),
+        annotations: tc.SArray(anns),
+    })
+    backgrounds = test[feature].head(5)
+    return train, test, backgrounds
+
+
+class OneObjectDetectorSmokeTest(unittest.TestCase):
+    @classmethod
+    def setUpClass(self):
+        """
+        The setup class method for the basic test case with all default values.
+        """
+        self.feature = 'myimage'
+        self.target = 'mytarget'
+        self.annotations = 'myannotations'
+
+        ## Create the model
+        self.def_opts = {
+           'model': 'darknet-yolo',
+           'max_iterations': 2,
+        }
+
+        # Model
+        self.train, self.test, self.backgrounds = _get_data(feature=self.feature, target = self.target, annotations=self.annotations)
+        self.model = tc.one_shot_object_detector.create(self.train,
+                                                        target=self.target,
+                                                        backgrounds=self.backgrounds,
+                                                        batch_size=2,
+                                                        max_iterations=1)
+
+        ## Answers
+        self.opts = self.def_opts.copy()
+        self.opts['max_iterations'] = 1
+
+        self.get_ans = {
+           'classes': lambda x: x == sorted(_CLASSES),
+           'target': lambda x: x == self.feature,
+           'num_starter_images': lambda x: x > 0,
+           'num_classes': lambda x: x == len(_CLASSES),
+        }
+        self.fields_ans = self.get_ans.keys()
+
+    @pytest.mark.xfail(raises = _ToolkitError)
```

Comment:
Use `self.assertRaises`, as in our other tests. `xfail` is not for tests that are "designed" to fail. See also https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions